### PR TITLE
Refactor Cloud Capabilities script for the Az module

### PR DESCRIPTION
**Pull request by mattbriggs**

### CODE REVIEW

File: Connect/AzureStack.Connect.psm1 [hunk 1]
```diff
@@ -1,7 +1,11 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # See LICENSE.txt in the project root for license information.
 
+<<<<<<< HEAD
+#requires -Modules VpnClient
+=======
 #requires -Modules Az.Accounts, VpnClient
+>>>>>>> 6b4244552743337e314e61f0aff2c5fb1a17e501
```
Comment (kilasuit) [inline Connect/AzureStack.Connect.psm1:8]: Some issue Git issue here that needs cleaning up

File: Registration/RegisterWithAzure.psm1 [hunk 2]
```diff
@@ -205,8 +244,14 @@ function Set-AzsRegistration{
...
     Log-Output "*********************** Begin log: $($PSCmdlet.MyInvocation.MyCommand.Name) ***********************`r`n"
 
+    Validate-AzureContext -AzureContext $AzureContext
     Validate-ResourceGroupLocation -ResourceGroupLocation $ResourceGroupLocation
+<<<<<<< HEAD
+    Validate-BillingModel -BillingModel $BillingModel -MsAssetTag $MsAssetTag
+    $azureAccountInfo = Get-AzureAccountInfo -AzureContext $AzureContext
+=======
     Log-AzureAccountInfo -AzureContext $AzureContext
+>>>>>>> 6b4244552743337e314e61f0aff2c5fb1a17e501
```
Comment (kilasuit) [inline Registration/RegisterWithAzure.psm1:254]: Same git issue here

File: Registration/RegisterWithAzure.psm1 [hunk 3]
```diff
@@ -335,7 +380,12 @@ function Remove-AzsRegistration{
 
     Log-Output "*********************** Begin log: $($PSCmdlet.MyInvocation.MyCommand.Name) ***********************`r`n"
 
+<<<<<<< HEAD
+    Validate-AzureContext -AzureContext $AzureContext
+    $azureAccountInfo = Get-AzureAccountInfo -AzureContext $AzureContext
+=======
     Log-AzureAccountInfo -AzureContext $AzureContext
+>>>>>>> 6b4244552743337e314e61f0aff2c5fb1a17e501
```
Comment (kilasuit) [inline Registration/RegisterWithAzure.psm1:388]: Another Git issue here

File: Registration/RegisterWithAzure.psm1 [hunk 4]
```diff
@@ -834,7 +887,12 @@ Function Get-AzsActivationKey{
...
```
Comment (kilasuit) [inline Registration/RegisterWithAzure.psm1:895]: Another git issue here

File: Registration/RegisterWithAzure.psm1 [hunk 5]
```diff
@@ -1787,11 +1846,53 @@ function Validate-ResourceGroupLocation{
         [Parameter(Mandatory=$true)]
         [string] $ResourceGroupLocation
     )
+<<<<<<< HEAD
```
Comment (kilasuit) [inline Registration/RegisterWithAzure.psm1:1849]: Another git issue here

File: Registration/RegisterWithAzure.psm1 [hunk 6]
```diff
@@ -1787,11 +1846,53 @@ function Validate-ResourceGroupLocation{
...
+        throw "ErrorCode: AzureContextNotSet.`nErrorReason: Azure Powershell context is null. Please log in to correct Azure Powershell context using 'Login-AzureRmAccount' and then call the registration cmdlet."
+    }
+}
+=======
     $availableLocations = (Get-AzLocation).Location
     if ($availableLocations -notcontains $ResourceGroupLocation){
         throw "ErrorCode: UnknownResourceGroupLocation.`nErrorReason: Resource group location '$ResourceGroupLocation' is not available. Please call the registration cmdlet along with ResourceGroupLocation parameter.`nAvailable locations: $($availableLocations -join ', ')`n"
     }
 }
+>>>>>>> 6b4244552743337e314e61f0aff2c5fb1a17e501
```
Comment (kilasuit) [inline Registration/RegisterWithAzure.psm1:1895]: Another git issue here

**Comment by mattbriggs** [comment]
@kilasuit Thanks for pointing that out. I resolved the conflicts. Thanks! Matt

**Comment by bganapa** [comment]
@mattbriggs  Thanks for the PR!, You have few commits that you dont want to be part of this PR

**Comment by mattbriggs** [comment]
@kilasuit and @bganapa updated. Thanks!

**Pull request merged**